tests-asyncio/test_asyncio_based_service.py

```diff
@@ -49,7 +49,10 @@ class WaitCancelledService(Service):
         # stopping.  Since it may be cancelled but still not stopped we
         # can't know.
         assert manager.is_cancelled is True
-        assert manager.is_stopping is True
+        # A service is only in the "stopping" phase for a very short period of
+        # time after all of the tasks have finished so we may either be
+        # stopping at this phase or completely finished.
+        assert manager.is_stopping is True or manager.is_finished is True
```

Comment:
this kind of suggests we have a "past tense" version of each of these states... by analogy to edge vs. level triggered w/ events, `is_stopping` during the transition and `is_stopped` after the event has occurred. in the normal case, you get nice structure like e.g. `is_finished` implies `is_stopped` implies `did_run` etc...

we may not currently have a need for this level of granularity in the lifecycle graph but just mentioning it given that the comment suggests it would already be helpful for the test

Reply:
I'm actually looking to remove the `stopping` APIs entirely.  Now that I've been back in the library and the tests I'm convinced they have little or no value and I completely agree with your assessment that this API diverges from the "philosophy" of the other ones.